hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/customfees/AssessmentResult.java

```diff
@@ -54,7 +54,11 @@ public AssessmentResult(
             final List<TokenTransferList> inputTokenTransfers, final List<AccountAmount> inputHbarTransfers) {
         mutableInputBalanceAdjustments = buildFungibleTokenTransferMap(inputTokenTransfers);
         immutableInputTokenAdjustments = Collections.unmodifiableMap(mutableInputBalanceAdjustments.entrySet().stream()
-                .collect(Collectors.toMap(Map.Entry::getKey, entry -> Map.copyOf(entry.getValue()))));
+                .collect(Collectors.toMap(
+                        Map.Entry::getKey,
+                        entry -> new LinkedHashMap<>(entry.getValue()),
```

Comment:
This will create a modifiable `Map`. Is this okay?

Reply:
Note this is wrapped in in `Collections.unmodifiableMap()`.

Reply:
Only the "outer" `Map` is wrapped. The "inner" `Map`s, created by `entry -> new LinkedHashMap<>(entry.getValue())`, are still modifiable.